### PR TITLE
Fixing mass migration

### PR DESCRIPTION
**Pull request by ChihChengLiang**
Some summary of the discussion today:

- no point checking spokeID in state transition, we should check it in dispute signature
- no point checking BURN_STATE_INDEX in the state transition, since we don't modify the receiver state
- no point checking tx.tokenType all equal, check them against the committed tokenType instead.

### CODE REVIEW

File: contracts/libs/Tx.sol [hunk 1]
```diff
@@ -278,4 +267,20 @@ library Tx {
...
         return txs.length / TX_LEN_5;
     }
+
+    function massMigration_messageOf(
+        MassMigration memory _tx,
+        uint256 nonce,
+        uint256 spokeID
+    ) internal pure returns (bytes memory) {
+        return
+            abi.encodePacked(
```
[the remark this answers is not in the document]
Comment (ChihChengLiang) [reply contracts/libs/Tx.sol:277]: I am now more convinced that we should use `encodePack` for the messages. The reason is when we do hash to field in BLS, there's a loop that iterates every 32 bytes we added to the raw message
https://github.com/thehubbleproject/hubble-contracts/blob/master/contracts/libs/BLS.sol#L338-L344

File: ts/tx.ts [hunk 2]
```diff
@@ -127,15 +127,17 @@ export class TxMassMigration implements SignableTx {
...
-            this.TX_TYPE,
-            hexZeroPad(hexlify(this.fromIndex), stateIDLen),
-            hexZeroPad(this.amount.toHexString(), 32),
-            hexZeroPad(this.fee.toHexString(), 32),
-            hexZeroPad(hexlify(this.nonce), nonceLen),
-            hexZeroPad(hexlify(this.spokeID), spokeLen)
-        ]);
-        return hexlify(concated);
+        return solidityPack(
+            ["uint8", "uint32", "uint256", "uint256", "uint32", "uint32"],
```
Comment (vaibhavchellani) [inline ts/tx.ts:131]: from index is `uint32` and to index is `uint256` not sure if it is a typo, just pointing out.

Comment (ChihChengLiang) [reply ts/tx.ts:131]: Oh, that's not a toIndex, we don't have the toIndex now.

Comment (vaibhavchellani) [reply ts/tx.ts:131]: Ah! Right, gotcha.

**Review by vaibhavchellani — APPROVED** [review]
looks good to me, might have some conflicts to resolve post merging #231

**Pull request merged**